### PR TITLE
ENH: Better logic and message

### LINKED ISSUE #10286: confusing warning with applying projs
#### Describe the bug
Using `apply_proj` after marking bad channels should automatically do the right thing instead of raising a confusing warning.

#### Steps to reproduce

```py
import os
import numpy as np
import mne

sample_data_folder = mne.datasets.sample.data_path()
sample_data_raw_file = os.path.join(sample_data_folder, 'MEG', 'sample',
                                    'sample_audvis_filt-0-40_raw.fif')
raw = mne.io.read_raw_fif(sample_data_raw_file)
raw = raw.pick_types(meg='mag', eeg=False)

print(raw.info['projs'])

raw.info['bads'] = raw.info['ch_names'][:16]
raw2 = raw.copy()

raw.apply_proj()
raw2.info.normalize_proj()
raw2.apply_proj()
np.testing.assert_allclose(raw.get_data(), raw2.get_data())
```

As per the warning, `normalize_proj` should be applied but as you can see, it makes no difference except getting rid of the warning.

#### Expected results
There should be no warning

#### Actual results
We see a warning:

```py
/Users/mainak/Desktop/mne_proj_issue.py:16: RuntimeWarning: Projection vector "PCA-v1" has magnitude 0.99 (should be unity), applying projector with 86/102 of the original channels available may be dangerous, consider recomputing and adding projection vectors for channels that are eventually used. If this is intentional, consider using info.normalize_proj()
  # raw.apply_proj()
/Users/mainak/Desktop/mne_proj_issue.py:16: RuntimeWarning: Projection vector "PCA-v2" has magnitude 0.95 (should be unity), applying projector with 86/102 of the original channels available may be dangerous, consider recomputing and adding projection vectors for channels that are eventually used. If this is intentional, consider using info.normalize_proj()
  # raw.apply_proj()
/Users/mainak/Desktop/mne_proj_issue.py:16: RuntimeWarning: Projection vector "PCA-v3" has magnitude 0.85 (should be unity), applying projector with 86/102 of the original channels available may be dangerous, consider recomputing and adding projection vectors for channels that are eventually used. If this is intentional, consider using info.normalize_proj()
```

#### Additional information
Additionally, the code path is somewhat confusing to follow. There is an `if inplace` clause:

https://github.com/mne-tools/mne-python/blob/93a3be5d21028331a88c42ef9dfa10566bcd6d1f/mne/io/proj.py#L618-L625

which makes the function return without performing an SVD. I see that `_make_projector` is never called with `inplace=False`. Is this dead code? We discussed this with @mshamalainen and the right thing to do is:

- combine the projectors into a matrix with bad channels zeroed out
- compute SVD of this matrix and use it to make the new projector

There is nothing "dangerous" per se in doing this as far as we can see. It only makes the projection operator mathematically correct. That is: [P.P = P](https://en.wikipedia.org/wiki/Projection_(linear_algebra)#Idempotence)

### PR DESCRIPTION
**Pull request by larsoner**
A concrete proposal for #10286:

1. Make the check depend on a 10%+ reduction in projector norm, which is better than looking for a 10% channel count reduction, and 
2. Improve the warning to make it clear that this can be problematic, or not

An alternative idea of adding a `suppress_proj_warning` or so has also been suggested, but I'm -1 on that because I think there are probably dozens of places we'd need to add it (i.e., anywhere that we apply proj -- creating epochs, creating evoked, plotting functions, cov computation, inv computation, ...).

Another alternative is to try to infer a possible problem during `compute_proj_*`, but I think that's going to be even more difficult given that some legitimate interference patterns are concentrated on a few channels. Warning later is admittedly a bit more annoying, but it also leverages additional information (i.e., that the user has potentially marked channels as bad now) that we don't have during the initial SSP computation. So I think the doc update here is enough to take care of the problem well enough.

@jasmainak would this have saved you some pain?

We could adapt my little example from #10286 into an example or tutorial, but I think it's a bit overkill...

Closes #10286

### CODE REVIEW

**Comment by jasmainak** [comment]
Fair enough, as I guess removing the warning is a no go. Why not rename `normalize_proj` at least since it gives the impression that SVD computation is done in that function and makes the code confusing to read?

We should develop some metric for detecting bad proj during fit but I guess that's more a science project

**Comment by larsoner** [comment]
> Why not rename normalize_proj at least since it gives the impression that SVD computation is done in that function and makes the code confusing to read?

There is a cost to renaming things, generally we try not to deprecate without really good reason. I don't think of SVD being done via the name `normalize`, why does it pop up for you?

**Comment by jasmainak** [comment]
Because it gives me the impression that the projection vectors are being rescaled to unit norm somehow as [done here](https://github.com/mne-tools/mne-matlab/blob/master/matlab/mne_make_projector.m#L130-L136)

**Comment by jasmainak** [comment]
Can you point me how `normalize_proj` does that?

**Comment by larsoner** [comment]
`normalize_proj` -> `_make_projector` with `inplace=True` to

https://github.com/mne-tools/mne-python/blob/258b462f06a3a5ef4c6d8ded6ca03f519b498831/mne/io/proj.py#L601-L616

And assignment

https://github.com/mne-tools/mne-python/blob/258b462f06a3a5ef4c6d8ded6ca03f519b498831/mne/io/proj.py#L621

You can see it in action in my example from #10286 with:
```
raw.info['bads'] = []
proj = mne.compute_proj_raw(raw, n_eeg=1)
raw.add_proj(proj)
print(raw.info['projs'][0]['data']['data'][0, :-1])
print(len(raw.info['projs'][0]['data']['col_names']))
raw.info['bads'] = [raw.ch_names[-1]]
raw.info.normalize_proj()
print(raw.info['projs'][0]['data']['data'][0])
print(len(raw.info['projs'][0]['data']['col_names']))
```
gives
```
[-1.80151819e-06  1.48807556e-05 -1.32953630e-05  6.50220538e-07
  2.19178494e-05  2.57570113e-06  5.37144244e-06  1.74230119e-05
  1.95853862e-05]
10
[-0.04496916  0.37145063 -0.33187635  0.01623068  0.54710924  0.06429417
  0.13408094  0.43490995  0.48888673]
9
```

**Comment by jasmainak** [comment]
okay fair enough!

**Comment by larsoner** [comment]
Okay for you @agramfort ?

File: mne/io/proj.py [hunk 1]
```diff
@@ -774,7 +806,7 @@ def make_eeg_average_ref_proj(info, activate=True, verbose=None):
         raise ValueError('Cannot create EEG average reference projector '
                          '(no EEG data found)')
     vec = np.ones((1, n_eeg))
-    vec /= n_eeg
+    vec /= np.sqrt(n_eeg)
```
Comment (agramfort) [inline mne/io/proj.py:809]: that's a bug fix?

**Review by agramfort — APPROVED** [review]
just a small comment. I don't get where this sqrt comes from.

File: mne/io/proj.py (hunk 1 above)
Comment (larsoner) [reply mne/io/proj.py:809]: Sqrt is really just for consistency. All other projectors we create are unit norm, and this makes the avg ref also have unit norm (which it doesn't on main).

When we actually compute the projection matrix, we renormalize (and then later orthonormalize again, I think), which is why it doesn't usually matter in practice, and hence doesn't really need to be called a bugfix. This change won't affect the user experience or outputs at all. But when doing low level internal testing, it's helpful to have all projectors created with unit norm in the first place.

**Pull request merged**